src/supervisor-api.ts

```diff
@@ -124,9 +77,19 @@ export class SupervisorAPI {
 		// Expires the supervisor's API key and generates a new one.
 		// It also communicates the new key to the balena API.
 		this.api.post('/v1/regenerate-api-key', async (_req, res) => {
-			const secret = config.newUniqueKey();
-			await config.set({ apiSecret: secret });
-			res.status(200).send(secret);
+			await deviceState.initialized;
+			await apiKeys.initialized;
+
+			// force a new cloud key
+			await apiKeys.generateCloudKey(true);
+
+			// report the new key to the cloud API
+			deviceState.reportCurrentState({
+				api_secret: apiKeys.cloudApiKey,
+			});
+
+			// return the value of the new key to the caller
+			res.status(200).send(apiKeys.cloudApiKey);
```

Comment:
Is this a possible security risk? The `cloudApiKey` is scoped to `global` so it can do anything, and it's returned to the client who has used a non-global scoped key. We should think about this @CameronDiver

Reply:
reading this: https://www.balena.io/docs/reference/supervisor/supervisor-api/#post-v1regenerate-api-key

... it seems there is 2 things going on here;
- the scoped key should be forcibly regenerated and returned
- the key is sent to the API

In the new scoped key world the key in the cloud is different to the key in the app. So I propose the following:
- regenerate the scoped key and return it to the client
- **IF** the key is the cloud key, send the key to the API also

Reply:
Sounds good to me